### PR TITLE
SIL: Perform move-only diag passes after ClosureLifetimeFixup.

### PR DESCRIPTION
**Pull request by jckarter**
ClosureLifetimeFixup finalizes the lifetimes of nonescaping closures, which allows us to analyze their captures as borrows and permit move- only types to be captured by them.

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

File: lib/SILOptimizer/PassManager/PassPipeline.cpp [hunk 1]
```diff
@@ -150,24 +171,6 @@ static void addMandatoryDiagnosticOptPipeline(SILPassPipelinePlan &P) {
...
-  P.addDifferentiation();
-
-  // Only run semantic arc opts if we are optimizing and if mandatory semantic
-  // arc opts is explicitly enabled.
-  //
-  // NOTE: Eventually this pass will be split into a mandatory/more aggressive
-  // pass. This will happen when OSSA is no longer eliminated before the
-  // optimizer pipeline is run implying we can put a pass that requires OSSA
-  // there.
-  const auto &Options = P.getOptions();
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassPipeline.cpp:168]: Can you add a test that shows this behavior? Then I am happy.

Comment (gottesmm) [reply lib/SILOptimizer/PassManager/PassPipeline.cpp:168]: I talked with JoeG offline about this. They mentioned that this is just the first part of a larger fix so it isn't time yet to add a test for this behavior. So... LGTM!

**Review by gottesmm — APPROVED** [review]

File: lib/SILOptimizer/PassManager/PassPipeline.cpp (hunk 1 above)
Comment (jckarter) [reply lib/SILOptimizer/PassManager/PassPipeline.cpp:168]: There is no behavior change yet.

**Comment by jckarter** [comment]
@swift-ci Please clean test

**Comment by jckarter** [comment]
@swift-ci Please clean test

**Pull request merged**